changebot/blueprints/circleci.py

```diff
@@ -0,0 +1,101 @@
+import json
+import requests
+from chagebot.github.github_auth import github_request_headers, get_installation_id_for_repo
+
+from flask import Blueprint, request, current_app
+
+circleci = Blueprint('circleci', __name__)
+
+
+HOST = "https://api.github.com"
+
+
+@circleci.route('/circleci', methods=['POST'])
+def circleci_handler():
+    if not request.data:
+        print("No payload received")
+
+    payload = json.loads(request.data)['payload']
+
+    # Validate we have the keys we need, otherwise ignore the push
+    required_keys = {'vcs_revision',
+                     'username',
+                     'reponame',
+                     'status',
+                     'build_num'}
+
+    if not required_keys.issubset(payload.keys()):
+        return 'Payload missing {}'.format(' '.join(required_keys - payload.keys()))
+
+    repo = f"{payload['username']}/{payload['reponame']}"
+
+    if payload['status'] == 'success':
+        artifacts = get_artifacts_from_build(payload)
+        url = get_documentation_url_from_artifacts(artifacts)
+
+        if url:
+            # Get installation id
+            installation_id = get_installation_id_for_repo(repo)
+            if not installation_id:
+                return f"Not installed for repo {repo}"
+            set_commit_status(repo, installation_id,
+                              payload['vcs_revision'], "success",
+                              "Click details to preview the documentation build", url)
+
+    return "All good"
+
+
+def get_artifacts_from_build(p):
+    base_url = "https://circleci.com/api/v1.1"
+    query_url = f"{base_url}/project/github/{p['username']}/{p['reponame']}/{p['build_num']}/artifacts"
+    response = requests.get(query_url)
+    assert response.ok, response.content
+    return response.json()
+
+
+def get_documentation_url_from_artifacts(artifacts):
+    for artifact in artifacts:
+        # Find the root sphinx index.html
+        if "html/index.html" in artifact['path']:
+            return artifact['url']
+
+
+def set_commit_status(repository, installation, commit_hash, state, description, target_url):
+
+    headers = github_request_headers(installation)
+
+    set_status(repository, commit_hash, state, description, current_app.name,
+               headers=headers, target_url=target_url)
+
+
+def set_status(repository, commit_hash, state, description, context, *, headers, target_url=None):
+    """
+    Set status message in a pull request on GitHub.
```

Comment:
@Cadair - why not use https://github.com/astropy/astropy-bot/blob/master/changebot/github/github_api.py#L466?

Reply:
This and that should probably be refactored so the code isn't duped. The reason I didn't use that method originally is that I am setting status on a *commit* not on a PR, so I didn't want to / couldn't create a PR object.

Reply:
While addressing @pllim's review, I've actually started that refactoring last night :)